src/test/java/edu/harvard/iq/dataverse/engine/command/impl/ReconcilePIDCommandTest.java

```diff
@@ -0,0 +1,354 @@
+
+        // given a dataset linked with a PidProvider and corresponding globalId
+        Dataset ds = MocksFactory.makeDataset();
+        // a
+        ds.setPidGenerator(pidProvider);
+
+        // given a linked dataset with a givendataverse and
+        Dataverse dv = MocksFactory.makeDataverse();
+        dv.setPidGenerator(pidProvider);
+        ds.setOwner(dv);
+
+
+
+        ReconcileDatasetPidCommand sut = new ReconcileDatasetPidCommand(request, ds, pidProvider);
+        assertThrows(IllegalCommandException.class, () -> engine.submit(sut),"Should throw ICE when no change");
+    }
+    @Test
+    public void testGuardAlreadyPublished() throws CommandException {
+        // given
+        //
+        AuthenticatedUser superUser = new AuthenticatedUser();
+        superUser.setSuperuser(true);
+        DataverseRequest request = MocksFactory.makeRequest(superUser);
+        // given a PID Provider
+        PidProvider pidProvider = mock(PidProvider.class);
+        when(pidProvider.getProtocol()).thenReturn("PROTO");
+        when(pidProvider.getAuthority()).thenReturn("PROTO");
+        when(pidProvider.getShoulder()).thenReturn("PROTO");
+        when(pidProvider.getSeparator()).thenReturn("PROTO");
+        when(pidProvider.registerWhenPublished()).thenReturn(true);
+
+        // given a dataset linked with the PIDprovider
+        Dataset ds = MocksFactory.makeDataset();
+        ds.setPidGenerator(pidProvider);
+        ds.setPublicationDate(Timestamp.from(Instant.now()));
+
+        // given a linked dataset with a givendataverse and
+        Dataverse dv = MocksFactory.makeDataverse();
+        dv.setPidGenerator(pidProvider);
+        ds.setOwner(dv);
+
+
+
+
+        ReconcileDatasetPidCommand sut = new ReconcileDatasetPidCommand(request, ds, pidProvider);
+        assertThrows(IllegalCommandException.class, () -> engine.submit(sut),"Should throw ICE when already published");
+    }
+    @Test
+    public void testGuardHarvested() throws CommandException {
+        // given
+        //
+        AuthenticatedUser superUser = new AuthenticatedUser();
+        superUser.setSuperuser(true);
+        DataverseRequest request = MocksFactory.makeRequest(superUser);
+        // given a PID Provider
+        PidProvider pidProvider = mock(PidProvider.class);
+        when(pidProvider.getProtocol()).thenReturn("PROTO");
+        when(pidProvider.getAuthority()).thenReturn("PROTO");
+        when(pidProvider.getShoulder()).thenReturn("PROTO");
+        when(pidProvider.getSeparator()).thenReturn("PROTO");
+        when(pidProvider.registerWhenPublished()).thenReturn(true);
+
+        // given a dataset linked with the PIDprovider
+        Dataset ds = MocksFactory.makeDataset();
+        ds.setHarvestedFrom(mock(HarvestingClient.class));
+        ds.setPidGenerator(pidProvider);
+
+        // given a linked dataset with a givendataverse and
+        Dataverse dv = MocksFactory.makeDataverse();
+        dv.setPidGenerator(pidProvider);
```

Comment:
This should only be needed if you make the change suggested earlier to drop the pidProvider param for the API call/command and instead pick up the dataset's effective Pid Provider, which, if it isn't set directly on the dataset will be the PidGenerator set on it's parent collection.